scripts/cli/src/util/sqlite3.ts

```diff
@@ -0,0 +1,39 @@
+import sql from "sqlite3";
+import type { KeyringPair } from "@polkadot/keyring/types";
+
+const sql3 = sql.verbose();
+// Setting up a database for storing data.
+const db = new sql3.Database("database.db");
+
+export function createTable() {
+  db.run(
+    "CREATE TABLE IF NOT EXISTS next_nonce ( account TEXT PRIMARY KEY, nonce INT DEFAULT 0 )"
+  );
+}
+
+export async function getNonce(signer: KeyringPair) {
+  return new Promise<number>((resolve, reject) => {
+    db.run(
+      "INSERT INTO next_nonce(account) VALUES($account) ON CONFLICT(account) DO UPDATE SET nonce=nonce+1",
```

Comment:
This looks like a race condition to me.

If one process gets a nonce, but is slow submitting the transaction, a second process could get the next nonce and submit before the first, leading to a potentially flaky test.

I think the simplest solution would be would be to add add retry logic to `sendTransaction` if we get a bad nonce error (I think its a '[Future](https://wiki.polkadot.network/docs/maintain-errors#common-errors)' error), since in this case the previous nonce should be submitted relatively soon.

Alternatively, we could wait and see if the race actually happens like I think it will too.

To actually coordinate the nonce between parallel processes gets more complicated.

Reply:
I have done tests with "future" nonces.  The node will hold on to the transaction and only execute it once the "missing" nonces have shown up.   (i.e. the node receives a tx `nonce=10`, but next nonce is 5, then receives txs with nonces 5-9, it will execute all transactions).

So it should be fine if a transaction is slow to arrive.  As long as that nonce gets "used".  Skipping a nonce value will stall execution of transactions for that account.

Reply:
Nice, I was thinking we would have to add in that logic ourselves.